### DISCUSSION TITLE
How to access frontmatter in a markdown-it plugin?

### DISCUSSION THREAD

**Discussion by Octobug** [vuejs/vitepress#3646] [Q&A] [answered]
This closed issue https://github.com/vuejs/vitepress/issues/1122 seems like what I need. But I cannot figure out how to realize it since VitePress has changed a lot from then on.

**Comment by brc-dd** [comment, marked as answer]
You can access it from `env` parameter in your renderer.

**Reply by Octobug** [reply]
Thank you! It took me some time to finally figure it out :D

**Comment by Octobug** [comment]
e.g.

```js
md.renderer.rules.footnote_block_open = (_tks, _idx, _opts, env) => {
  const { frontmatter } = env;
  const title = frontmatter.footnote || "References";
  ...
};
```
